pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -210,11 +234,9 @@ func (a *ipAssigner) AssignedIPs() sets.String {
 
 // Run starts the ARP responder and NDP responder.
 func (a *ipAssigner) Run(ch <-chan struct{}) {
-	// Start the ARP responder only when the dummy device is not created. The kernel will handle ARP requests
-	// for IPs assigned to the dummy devices by default.
-	// TODO: Check the arp_ignore sysctl parameter of the transport interface to determine whether to start
-	// the ARP responder or not.
-	if a.dummyDevice == nil && a.arpResponder != nil {
+	// Start the ARP responder if the dummy device does not exist or
+	// arp_ignore of the transport interface has value other than 0.
+	if (a.arpIgnore > 0 || a.dummyDevice == nil) && a.arpResponder != nil {
```

Comment:
Should we move this to intialization: create arpResponder only when it's required, and run it when it exists?
To avoid creating an instance when we don't use it and save a member variable.